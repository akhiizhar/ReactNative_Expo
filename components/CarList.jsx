import { Text, StyleSheet, Image, Pressable, style } from "react-native";
import React from "react";
import { Col, Row } from "@/components/Grid";
import Ionicons from "@expo/vector-icons/Ionicons";

const formatCurrency = new Intl.NumberFormat("id-ID", {
	style: "currency",
	currency: "IDR",
});
export default function CarList({
	onPress,
	image,
	carName,
	passenger,
	baggage,
	price,
}) {
	return (
		<Pressable style={{ ...styles.card, ...style }} onPress={onPress}>
			<Row alignItems={"center"} gap={15}>
				<Col>
					<Image style={styles.img} source={image} />
				</Col>
				<Col>
					<Text style={styles.carName}>{carName}</Text>
					<Row>
						<Col style={styles.textIcon}>
							<Ionicons size={24} name={"people-outline"} color={"#8A8A8A"} />
							<Text style={styles.capacityText}>{passenger}</Text>
						</Col>
						<Col style={styles.textIcon}>
							<Ionicons size={24} name={"bag-outline"} color={"#8A8A8A"} />
							<Text style={styles.capacityText}>{baggage}</Text>
						</Col>
					</Row>
					<Text style={styles.price}>{formatCurrency.format(price)}</Text>
				</Col>
			</Row>
		</Pressable>
	);
}

const styles = StyleSheet.create({
	card: {
		borderColor: "rgba(0,0,0,0.2)",
		borderWidth: 0.5,
		borderRadius: 4,
		padding: 5,
		marginBottom: 20,
		marginVertical: 20,
		marginHorizontal: 20,
	},
	img: {
		width: 80,
		height: 80,
		// marginTop: 0,
		// marginLeft: 15,
		margin: 5,
		objectFit: "contain",
	},
	carName: {
		fontSize: 16,
		marginTop: 20,
		fontFamily: "PoppinsRegular",
	},
	capacityText: {
		color: "#8A8A8A",
		fontSize: 16,
		fontFamily: "PoppinsBold",
	},
	price: {
		color: "#5CB85F",
		marginBottom: 20,
		marginTop: 10,
		fontSize: 16,
		fontFamily: "PoppinsRegular",
	},
	textIcon: {
		flexDirection: "row",
		alignItems: "center",
		gap: 2,
		margin: 5,
	},
});
