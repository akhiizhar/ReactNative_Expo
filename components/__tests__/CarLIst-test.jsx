import * as React from "react";
import renderer from "react-test-renderer";

import CarList from "../CarList";

it(`CarList renders correctly`, () => {
	const tree = renderer
		.create(
			<CarList
				image={require("@/assets/images/img_photo.png")}
				carName={"Innova"}
				passengers={5}
				baggage={4}
				price={"200000"}
			/>
		)
		.toJSON();

	expect(tree).toMatchSnapshot();
});
