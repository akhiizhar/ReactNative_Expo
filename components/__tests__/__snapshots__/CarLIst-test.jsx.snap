// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarList renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "rgba(0,0,0,0.2)",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "marginBottom": 20,
      "marginHorizontal": 20,
      "marginVertical": 20,
      "padding": 5,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 15,
        "justifyContent": "flex-start",
      }
    }
  >
    <View>
      <Image
        source={1}
        style={
          {
            "height": 80,
            "margin": 5,
            "objectFit": "contain",
            "width": 80,
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          {
            "fontFamily": "PoppinsRegular",
            "fontSize": 16,
            "marginTop": 20,
          }
        }
      >
        Innova
      </Text>
      <View
        style={
          {
            "alignItems": "baseline",
            "flexDirection": "row",
            "gap": 0,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 2,
              "margin": 5,
            }
          }
        >
          <Text />
          <Text
            style={
              {
                "color": "#8A8A8A",
                "fontFamily": "PoppinsBold",
                "fontSize": 16,
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 2,
              "margin": 5,
            }
          }
        >
          <Text />
          <Text
            style={
              {
                "color": "#8A8A8A",
                "fontFamily": "PoppinsBold",
                "fontSize": 16,
              }
            }
          >
            4
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#5CB85F",
            "fontFamily": "PoppinsRegular",
            "fontSize": 16,
            "marginBottom": 20,
            "marginTop": 10,
          }
        }
      >
        Rp 200.000,00
      </Text>
    </View>
  </View>
</View>
`;

exports[`CarListrenders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "rgba(0,0,0,0.2)",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "marginBottom": 20,
      "marginHorizontal": 20,
      "marginVertical": 20,
      "padding": 5,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 15,
        "justifyContent": "flex-start",
      }
    }
  >
    <View>
      <Image
        source={1}
        style={
          {
            "height": 80,
            "margin": 5,
            "objectFit": "contain",
            "width": 80,
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          {
            "fontFamily": "PoppinsRegular",
            "fontSize": 16,
            "marginTop": 20,
          }
        }
      >
        Innova
      </Text>
      <View
        style={
          {
            "alignItems": "baseline",
            "flexDirection": "row",
            "gap": 0,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 2,
              "margin": 5,
            }
          }
        >
          <Text />
          <Text
            style={
              {
                "color": "#8A8A8A",
                "fontFamily": "PoppinsBold",
                "fontSize": 16,
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 2,
              "margin": 5,
            }
          }
        >
          <Text />
          <Text
            style={
              {
                "color": "#8A8A8A",
                "fontFamily": "PoppinsBold",
                "fontSize": 16,
              }
            }
          >
            4
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#5CB85F",
            "fontFamily": "PoppinsRegular",
            "fontSize": 16,
            "marginBottom": 20,
            "marginTop": 10,
          }
        }
      >
        Rp 200.000,00
      </Text>
    </View>
  </View>
</View>
`;
